OpenRA.Game/MiniYaml.cs

```diff
@@ -136,19 +159,23 @@ public MiniYaml Clone()
 			return ret;
 		}
 
-		public MiniYaml(string value)
-			: this(value, null) { }
-
-		public MiniYaml(string value, List<MiniYamlNode> nodes)
+		public bool TryGetNodes(string key, out List<MiniYamlNode> value)
 		{
-			Value = value;
-			Nodes = nodes ?? new List<MiniYamlNode>();
-		}
+			if (key == null)
+			{
+				value = null;
+				return false;
+			}
 
-		public static List<MiniYamlNode> NodesOrEmpty(MiniYaml y, string s)
-		{
-			var nd = y.ToDictionary();
-			return nd.ContainsKey(s) ? nd[s].Nodes : new List<MiniYamlNode>();
+			foreach (var y in Nodes)
+				if (key == y.Key)
+				{
+					value = y.Value != null ? y.Value.Nodes : null;
+					return true;
+				}
```

Comment:
```suggestion
			foreach (var y in Nodes)
			{
				if (key == y.Key)
				{
					value = y.Value != null ? y.Value.Nodes : null;
					return true;
				}
			}
```
(Our coding style requires braces if the inner statement spans multiple lines.)
Btw, is it possible to replace `y.Value != null ? y.Value.Nodes : null;` with `y.Value?.Nodes`?

Reply:
👍 for `y.Value?.Nodes`.